pkgs/build-support/rust/rustc-wrapper/default.nix

```diff
@@ -46,24 +52,50 @@ runCommand "${rustc-unwrapped.pname}-wrapper-${rustc-unwrapped.version}"
         targetPlatforms
         badTargetPlatforms
         ;
-      unwrapped = rustc-unwrapped;
+      unwrapped = package;
     };
 
-    meta = rustc-unwrapped.meta // {
-      description = "${rustc-unwrapped.meta.description} (wrapper script)";
+    meta = package.meta // {
+      description = "${package.meta.description} (wrapper script)";
       priority = 10;
     };
   }
-  ''
-    mkdir -p $out/bin
-    ln -s ${rustc-unwrapped}/bin/* $out/bin
-    rm $out/bin/{rustc,rustdoc}
-    prog=${rustc-unwrapped}/bin/rustc extraFlagsVar=NIX_RUSTFLAGS \
-        substituteAll ${./rustc-wrapper.sh} $out/bin/rustc
-    prog=${rustc-unwrapped}/bin/rustdoc extraFlagsVar=NIX_RUSTDOCFLAGS \
-        substituteAll ${./rustc-wrapper.sh} $out/bin/rustdoc
-    chmod +x $out/bin/{rustc,rustdoc}
-    ${lib.concatMapStrings (output: "ln -s ${rustc-unwrapped.${output}} \$${output}\n") (
-      lib.remove "out" rustc-unwrapped.outputs
-    )}
-  ''
+  (
+    if flavor == "rustc" then
+      ''
+        mkdir -p $out/bin
+        ln -s ${package}/bin/* $out/bin
+        rm $out/bin/{rustc,rustdoc}
+        prog=${package}/bin/rustc extraFlagsVar=NIX_RUSTFLAGS \
+            substituteAll ${./rustc-wrapper.sh} $out/bin/rustc
+        prog=${package}/bin/rustdoc extraFlagsVar=NIX_RUSTDOCFLAGS \
+            substituteAll ${./rustc-wrapper.sh} $out/bin/rustdoc
+        chmod +x $out/bin/{rustc,rustdoc}
+        ${lib.concatMapStrings (output: "ln -s ${package.${output}} \$${output}\n") (
+          lib.remove "out" package.outputs
+        )}
+      ''
+    else if flavor == "clippy" then
```

Comment:
I'd like to see these deduplicated further. The clippy-specific PATH wrap shouldn't be in here (do we need it at all given we didn't have it before?), but we could loop through all of `rustc rustdoc clippy-driver` and wrap each one that exists.